### PR TITLE
Remove Singular dependencies from msolve

### PR DESCRIPTION
**Pull request by ederc**
In order to prepare thread-safe f4 based multi-modular code in Oscar.

### CODE REVIEW

File: src/Rings/msolve/interface.jl [hunk 1]
```diff
@@ -2,6 +2,77 @@
...
+            append!(exps, convert(Vector{Int32}, ev))
+        end
+    end
+
+    return lens, cfs, exps
+end
+
+function convert_ff_gb_array_to_oscar_array(
+        bld::Int32,
+        blen::Array{Int32,1},
```
Comment (fingolfin) [inline src/Rings/msolve/interface.jl]:
```suggestion
        blen::Vector{Int32},
```
We usually write `Vector` these days (and `Matrix{T}` for `Array{T,2}`), as this is what Julia does sine 1.6. However, this is not a blocker for this PR, I just wanted to point it out, feel free to change it later / never

File: src/Rings/msolve/interface.jl [hunk 2]
```diff
@@ -2,6 +2,77 @@
...
+        for i in 1:nr_gens
+            for cf in coefficients(I[i])
+                push!(cfs, BigInt(numerator(cf)))
+                push!(cfs, BigInt(denominator(cf)))
+            end
+        end
+    else
+        for i in 1:nr_gens
+            for cf in coefficients(I[i])
+                push!(cfs, Int32(cf.data))
```
Comment (fingolfin) [inline src/Rings/msolve/interface.jl:34]: Just so I understand: this makes some specific assumptions about the coefficients, namely that they are from a prime field with small modulus, right? Is this checked or enforced anywhere?

**Review by fingolfin — APPROVED** [review]

File: src/Rings/msolve/interface.jl (hunk 2 above)
Comment (ederc) [reply src/Rings/msolve/interface.jl:34]: Good remark, I've added a check for the size of the prime characteristic.

File: src/Rings/msolve/interface.jl [hunk 3]
```diff
@@ -2,13 +2,90 @@
...
+    end
+
+    return lens, cfs, exps
+end
+
+function convert_ff_gb_array_to_oscar_array(
+        bld::Int32,
+        blen::Vector{Int32},
+        bcf::Array{Int32,1},
+        bexp::Array{Int32,1},
```
Comment (fingolfin) [inline src/Rings/msolve/interface.jl]:
```suggestion
        bcf::Vector{Int32},
        bexp::Vector{Int32},
```

**Review by fingolfin — APPROVED** [review]

**Pull request merged**